### ISSUE TITLE
when the latent factors is equal to 8 or 16, the HR and NDCG is equal to 1.0

### ISSUE THREAD

**Issue by fjssharpsword** [RuidongZ/Deep_Matrix_Factorization_Models#6]
Problem:
When I sets the number of latent factors is 8 or 16, the experimental results of HR and NDCG all are 1.0. It is not reasonable. 
`parser.add_argument('-userLayer', action='store', dest='userLayer', default=[512, 8])
    parser.add_argument('-itemLayer', action='store', dest='itemLayer', default=[1024, 8])`

By debugging, I finds that the self.y_ always is 1e-6.
`self.y_ = tf.reduce_sum(tf.multiply(user_out, item_out), axis=1, keepdims=False) / (norm_item_output* norm_user_output)
        self.y_ = tf.maximum(1e-6, self.y_)`

In your paper, the experimental results showed that the HR and NDCG is normal when the number of latent factors is 8 or 16, can you give me the solution, thanks!
